File: C++/Enemy.cpp
#include "Enemy.h"

Enemy::Enemy() : Entity() {
	Name = "Zombie";
	Health = 10;
	Defense = 1;
	Level = 1;
	Agression = 100;
	XPValue = 51;
	Attack = 10;
	Symbol = 'Z';
}

Enemy::Enemy(int health, int attack, int defense, int level, int XPValue, int Agression, std::string Name, char Symbol) 
	: Entity(Name, health, defense, attack, level, Symbol), XPValue(), Agression() {}

void Enemy::onLevelUp() {
	Attack += 1;
	Health += 5;
	XPValue += 50;
}

Enemy& Enemy::operator=(const Enemy& other) {
	if (this == &other) {
		return *this;
	}
	Entity::operator=(other); // Копируем базовую часть
	XPValue = other.XPValue;
	Agression = other.Agression;
	return *this;
}

std::ostream& operator<<(std::ostream& os, const Enemy& enemy) {
	os << static_cast<const Entity&>(enemy);
	os << "XP value: " << enemy.XPValue << "\n";
	return os;
}

int Enemy::getLevel() {
	return Level;
}

int Enemy::getXPValue() {
	return XPValue;
}

bool Enemy::operator==(const Enemy& other) const {
	return this->getPosition() == other.getPosition();
}

File: Java/Enemy.java
package Java;

public class Enemy extends Entity implements Killable{
    private int xpValue;

    public Enemy() {
        super("Zombie", 10, 1, 7, 1, 'Z'); // Инициализация через конструктор родительского класса
        this.xpValue = 50;
    }

    public Enemy(int health, int attack, int defense, 
                int level, int xpValue, 
                String name, char symbol) {
        super(name, health, defense, attack, level, symbol);
        this.xpValue = xpValue;
    }

    @Override
    public int getHealth() {
        return health;
    }

    @Override
    public boolean isAlive() {
        return health > 0;
    }

    @Override
    protected void onLevelUp() {
        this.health += 5; 
        this.defense += 1;
        this.xpValue += 50;
    }

    public int getLevel() {
        return level;
    }

    public int getXPValue() {
        return xpValue;
    }

    @Override
    public String toString() {
        return super.toString() + 
               "XP value: " + xpValue + "\n";
    }
}
